src/data/nat/fib/zeckendorf.lean

```diff
@@ -0,0 +1,229 @@
+/-
+Copyright (c) 2022 Jakub Kądziołka. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakub Kądziołka
+-/
+import data.nat.fib.basic
+import algebra.big_operators.order
+import tactic.omega.main
+import tactic.linarith
+
+/-!
+# Zeckendorf's Theorem
+
+`nat.zeckendorf`: Each natural number can be written in exactly one way as a sum of distinct,
+non-consecutive Fibonacci numbers.
+-/
+
+open_locale big_operators
+
+namespace nat
+
+/-- The set `S` contains the indices which have been chosen.
+    For uniqueness, we need to exclude `fib 0` and `fib 1`,
+    since `fib 0 = 0` and `fib 1 = fib 2 = 1`. -/
+def is_zeckendorf_rep (S : finset ℕ) : Prop :=
+  ∀ (n ∈ S), n ≥ 2 ∧ n + 1 ∉ S
+
+/-- Every natural number lies between two Fibonacci numbers. -/
+lemma fib_between (n : ℕ): ∃ k : ℕ, fib k ≤ n ∧ n < fib k.succ :=
+begin
+  induction n using nat.two_step_induction with n IHn IHnsucc,
+  { use 0, simp },
+  { use 2, simp },
+  { obtain ⟨k, Hfibk_around_nsucc⟩ := IHnsucc,
+    by_cases Hfib_ksucc : n.succ.succ = fib k.succ,
+    { use k.succ,
+      have k_not_0 : k ≠ 0,
+      { assume H : k = 0,
+        have bad := calc n.succ.succ = fib k.succ : Hfib_ksucc
+                                 ... = fib 1      : by simp [H]
+                                 ... = 1          : rfl,
+        cases bad },
+      split,
+      show fib k.succ ≤ n.succ.succ, by simp [Hfib_ksucc],
+      show n.succ.succ < fib k.succ.succ,
+      { rw Hfib_ksucc, apply fib_lt_fib_succ, omega } },
```

Comment:
Try not to use `omega`. It's being phased out.

Reply:
Is there another tactic that understands that if `n <= k`, then `n < k.succ`?

Reply:
It's a lemma, `nat.lt_succ_iff`, and it's more generally true in a `succ_order`. You don't need tactics for everything.